### PR TITLE
8314838: 3 compiler tests ignore vm flags

### PR DESCRIPTION
**Pull request by lepestock**
For the compiler tests in question this PR:

- replaces all calls to `ProcessTools.createJavaProcessBuilder` with another method, the `createTestJvm`, which prepends given arguments with what was passed to the enclosing test. The change therefore...
- makes sure the external arguments aren't ignored and are used for creating the inner JVM;
- adds `@requires` guards in some places - where the test explicitly controls the Tiered Compilation, for example.

Testing: a local and a CI run, no failures registered.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8314838](https://bugs.openjdk.org/browse/JDK-8314838): 3 compiler tests ignore vm flags (**Sub-task** - P5)

### Reviewers
 * [Evgeny Astigeevich](https://openjdk.org/census#eastigeevich) (@eastig - Committer) ⚠️ Review applies to [cd5706e5](https://git.openjdk.org/jdk/pull/15609/files/cd5706e546f5a8f9fdee87e7db22a040524bcf45)
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**)
 * [Leonid Mesnik](https://openjdk.org/census#lmesnik) (@lmesnik - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15609/head:pull/15609` \
`$ git checkout pull/15609`

Update a local copy of the PR: \
`$ git checkout pull/15609` \
`$ git pull https://git.openjdk.org/jdk.git pull/15609/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15609`

View PR using the GUI difftool: \
`$ git pr show -t 15609`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15609.diff">https://git.openjdk.org/jdk/pull/15609.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15609#issuecomment-1709645281)

### CODE REVIEW

**Review by eastig — APPROVED** [review]
lgtm

File: test/hotspot/jtreg/compiler/sharedstubs/SharedStubToInterpTest.java [hunk 1]
```diff
@@ -23,15 +23,21 @@
...
 
 /**
- * @test SharedStubToInterpTest
- * @summary Checks that stubs to the interpreter can be shared for static or final method.
+ * @test id=C1
  * @bug 8280481
+ * @summary Checks that stubs to the interpreter can be shared for static or final method.
  * @library /test/lib
+ * @requires vm.opt.TieredStopAtLevel == null & vm.opt.TieredCompilation == null
+ * @requires os.arch=="amd64" | os.arch=="x86_64" | os.arch=="i386" | os.arch=="x86" | os.arch=="aarch64" | os.arch=="riscv64"
```
Comment (vnkozlov) [inline test/hotspot/jtreg/compiler/sharedstubs/SharedStubToInterpTest.java:31]: Consider using `vm.simpleArch` in `@requires`:
```
@requires vm.simpleArch == "x86" | vm.simpleArch == "x64" | vm.simpleArch == "aarch64" | vm.simpleArch == "riscv64"
```

Comment (lepestock) [reply test/hotspot/jtreg/compiler/sharedstubs/SharedStubToInterpTest.java]: Fixed. Thanks Vladimir!

**Review by vnkozlov — APPROVED** [review]
Good.

**Review by lmesnik — APPROVED** [review]

**Pull request closed without merging**